saplings/circuits/src/components/forms/propose_circuit/service.js

```diff
@@ -14,565 +14,442 @@
-          <div className="field-input">
-            <input
-              className="service-input"
-              value={serviceState.service.serviceType}
-              onChange={e => {
-                serviceDispatcher({
-                  type: 'set-service-type',
-                  serviceType: e.target.value
-                });
-              }}
-              disabled={!editMode}
+    <div className="form service-form">
+      <span className="service-form-header">Add a Service</span>
+      <div className="form-content">
+        <InputRow>
+          {basicInput('service-type', 'Service Type')}
+          {basicInput(
+            'service-id',
+            'Service ID',
+            <Button
+              label={<FontAwesomeIcon icon={faSync} />}
+              className="icon-button"
+              onClick={generateId}
             />
-            <div className="form-error">{serviceState.errors.serviceType}</div>
-          </div>
-        </div>
-        <div className="service-field bg-color-grey">
-          <div className="field-name">Service ID</div>
-          <div className="field-input">
-            <div className="flex-input">
-              <input
-                className="service-input service-id-input"
-                value={serviceState.service.serviceId}
-                onChange={e => {
-                  serviceDispatcher({
-                    type: 'set-service-id',
-                    serviceId: e.target.value
-                  });
-                }}
-                maxLength="4"
-                disabled={!editMode}
-              />
-              {editMode && (
-                <button
-                  className="icon-btn"
-                  type="button"
-                  onClick={() => {
-                    const id = generateServiceID();
-                    serviceDispatcher({
-                      type: 'set-service-id',
-                      serviceId: id
-                    });
-                  }}
-                  title="Generate service ID"
-                >
-                  <FontAwesomeIcon icon="sync-alt" />
-                </button>
-              )}
-            </div>
-            <div className="form-error">{serviceState.errors.serviceId}</div>
-          </div>
-        </div>
-        <div className="service-field">
-          <div className="field-name">Allowed node</div>
-          <div className="field-input">
-            {!editMode && allowedNodes}
-            {editMode && allowedNodeEditMode}
-            <div className="form-error">{serviceState.errors.allowedNodes}</div>
-          </div>
-        </div>
-        <div className="service-field bg-color-grey">
-          <div className="field-name">Service arguments</div>
-          <div className="field-input service-arguments">
-            {serviceArguments()}
-          </div>
+          )}
+        </InputRow>
+        <InputRow>
+          <ListBoxSelect
```

Comment:
I think that the text box of this should align with the edge of the other text boxes, rather than the buttons

Reply:
I agree, I think it would be more clear if the "generate id" button was part of the control, instead of next to it.  I'll see what I can do here.

Reply:
I've moved the button into the field, so it is more connected to the control.